services/ymax-planner/src/kv-store.ts

```diff
@@ -0,0 +1,86 @@
+/**
+ * Key-value store interface for storing and retrieving configuration values.
+ * This abstraction allows for different storage backends to be plugged in.
+ */
+import type { KVStore } from '@agoric/internal';
+import { makeKVStore } from '@agoric/internal';
+import type { Database as SQLiteDatabase } from 'better-sqlite3';
+import Database from 'better-sqlite3';
+
+export type SQLiteKeyValueStoreOptions = {
+  trace: (...args: string[]) => void;
+  tableName?: string;
+};
+
+export const makeSQLiteKeyValueStore = (
+  dbPath: string,
+  options: SQLiteKeyValueStoreOptions,
+): {
+  db: SQLiteDatabase;
+  kvStore: KVStore;
+} => {
+  const { tableName = 'kv_store' } = options;
+
+  // Note that the `dbPath` file is created if necessary:
+  // https://github.com/WiseLibs/better-sqlite3/blob/HEAD/docs/api.md#new-databasepath-options
+  const db = new Database(dbPath);
+
+  db.pragma('journal_mode = WAL');
+  db.pragma('synchronous = NORMAL');
+  db.pragma('busy_timeout = 5000');
+
+  if (!/^[A-Za-z_][A-Za-z0-9_]*$/.test(tableName)) {
+    throw new Error(`Invalid table name: ${tableName}`);
+  }
+
+  /**
+   * @see https://github.com/Agoric/agoric-sdk/blob/a62e38f05bf0b676c708b7861b88b2b70b841469/packages/swing-store/src/swingStore.js#L274-L287
+   * ensureTx is not required here since we want each write to the db
+   * to be committed instantly
+   */
+  const emptyEnsure = () => null;
+  const kvStore = makeKVStore(db, emptyEnsure, options.trace, tableName);
+  return { db, kvStore };
+};
+
+const getResolverLastBlockKey = (txId: `tx${number}`, suffix?: string) =>
+  `RESOLVER_LAST_BLOCK_${txId}${suffix ? `_${suffix}` : ''}`;
```

Comment:
Why capitalize the prefix for variable data, or for that matter have a prefix that will result in non-contiguous keys relating to a single transaction if we ever add other ?
```suggestion
const getTxBlockLowerBoundKey = (txId: `tx${number}`, scope?: string) =>
  `${txId}.blockLowerBound${scope ? `_${scope}` : ''}`;
```